### ISSUE TITLE
efficiently load entire model from DB to memory

### ISSUE THREAD

**Issue by 2x2xplz** [Raizlabs/DBFlow#1249]
DBFlow Version: 4.0.0-beta7
Kotlin 1.1
Issue Kind (Bug, Question, Feature): Question

Please note if you are using Instant Run, there may be bugs where generated classes are not created. Ensure you are using the [apt](https://bitbucket.org/hvisser/android-apt) or [kapt](http://blog.jetbrains.com/kotlin/2015/06/better-annotation-processing-supporting-stubs-in-kapt/) plugins and that incremental compilation is off.

Description: 
My app has a LEAGUE model which holds a list of TEAMS, each of which contains a list of PLAYERS. My TEAM class contains a getMyPlayers() function annotated with @OneToMany() for Method.LOAD, while the PLAYER class contains a TEAM variable annotated with @ForeignKey(), as per the [documentation](https://github.com/Raizlabs/DBFlow/blob/master/usage2/Relationships.md#one-to-many)

(I am using Kotlin, so the syntax is slightly different)

When the app starts, the entire LEAGUE model should get loaded into memory. It runs:
    
    for (team in SQLite.select().from(Team::class.java).queryList()) {
      League.allTeams.add(team)
    }

Because of the getMyPlayers() function (i.e. like getMyAnts() ), the above code automatically populates the Player list in each Team. However, it is inefficient, as it runs a SELECT for every team. A more efficient pattern would be like below, just one SELECT from Players, which reduced the load time by 80+% (I commented out the @OneToMany() block). 

    val playerList = SQLite.select().from(Player::class.java).queryList()
    for (team in SQLite.select().from(Team::class.java).queryList()) {
      for (player in playerList.filter { it.myTeam.teamID == team.teamID } ) {
        team.Players.add(player)
      }
      League.allTeams.add(team)
    }

However it seemed the app was not reliably aware of the relationships, I think because the "relationship" was assigned outside of DBFlow (sometimes a player could not say what team he belonged to, or vice versa -- especially after terminating the app then restarting, requiring a reload from DB).

Bottom line: is there a way to tell DBFlow to efficiently load the entire model into memory, while it maintains all the correct relationship information? The difference is an app startup time of 4-6 seconds vs under 1 second. Thank you.

**Comment by agrosner** [comment]
dont use `@OneToMany` on significant loads. The relationship for that is a fake one - i.e. inserted as a convenience for the processor. you can still call `getMyPlayers()` manually, itll just happen when you call it.
